cli/src/commands/absorb.rs

```diff
@@ -0,0 +1,1172 @@
+// Copyright 2024 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::cmp;
+use std::collections::HashMap;
+use std::io::Read;
+use std::ops::Range;
+use std::rc::Rc;
+
+use bstr::BString;
+use futures::StreamExt as _;
+use itertools::Itertools as _;
+use jj_lib::annotate::get_annotation_with_file_content;
+use jj_lib::backend::BackendError;
+use jj_lib::backend::BackendResult;
+use jj_lib::backend::CommitId;
+use jj_lib::backend::FileId;
+use jj_lib::backend::TreeValue;
+use jj_lib::commit::Commit;
+use jj_lib::conflicts::materialized_diff_stream;
+use jj_lib::conflicts::MaterializedTreeValue;
+use jj_lib::copies::CopyRecords;
+use jj_lib::diff::Diff;
+use jj_lib::diff::DiffHunkKind;
+use jj_lib::matchers::Matcher;
+use jj_lib::merge::Merge;
+use jj_lib::merged_tree::MergedTree;
+use jj_lib::merged_tree::MergedTreeBuilder;
+use jj_lib::repo::MutableRepo;
+use jj_lib::repo::Repo;
+use jj_lib::repo_path::RepoPath;
+use jj_lib::repo_path::RepoPathUiConverter;
+use jj_lib::revset::ResolvedRevsetExpression;
+use jj_lib::settings::UserSettings;
+use pollster::FutureExt as _;
+use tracing::instrument;
+
+use crate::cli_util::CommandHelper;
+use crate::cli_util::RevisionArg;
+use crate::command_error::CommandError;
+use crate::ui::Ui;
+
+/// Move changes from a revision into the stack of draft revisions
+///
+/// This command splits changes in the source revision and moves each change to
+/// the revision where the corresponding lines modified last. If the destination
```

Comment:
nit: "the revision" -> "the closest mutable ancestor" or something like that to be more specific